update

```diff
@@ -59,19 +65,35 @@ do_fixedref()
     )
 }
 
-checkout_ffmpeg=do_gitmaster
-#checkout_fribidi=do_releasetag
-checkout_libass=do_gitmaster
-checkout_mpv=do_gitmaster
+# args: $1: project name, $2: release/master [,$3: non-empty to ignore the config file]
```

Comment:
why not
* $2 non-empty -> use that
* $2 empty -> use config
    * config empty -> use master

The only downside would be that you can't then set different defaults on different packages, but I don't think we really care about that now since it's all master. If that ever changes, I think $2 should be the default and $3 should be empty or override value.

Reply:
> why not
> 
>     $2 non-empty -> use that
>     $2 empty -> use config
>         config empty -> use master

Because this function is called with one of two use cases (which I didn't modify):
- Try the config file, and if it doesn't exist then use the provided value.
- Just use the provided value.

And the API supports exactly these two cases, while your suggestion does not.

Also, `master` as fallback happens to be the current _choice_ as default value for all projects, but IMO it's not something which should be taken for granted by disallowing any other default values, which your suggestion would do.

In general, I think your approach would make the code more implicit in a way which is less easy to modify further. My goal was not to reduce the number of chars at the script, but rather make it easier to modify further.

Reply:
Ok, I just felt kind of weird about the "-" argument.

Reply:
> Ok, I just felt kind of weird about the "-" argument.

I completely agree about this. One of previous versions of this API had $2 as a config file name to try first, but ultimately I decided that the config file is always derived from the name, so it's more correct to indicate whether or not to use a config file rather than providing or not providing a redundant config file name.

It's not too bad, but I agree it's not amazing either. But it's better than before, which was the main goal ;)